src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogPostForMailItem.java

```diff
@@ -1,29 +1,110 @@
-        this.title = blogPost.getTitle();
-        this.url = blogPost.getUrl();
-        this.authorLabel = determineAuthorLabel(blogPost.getBlog());
+    /**
+     * Returns internal builder.
+     * 
+     * @return builder
+     */
+    public static Builder builder() {
+        return new Builder();
     }
+    
+    /**
+     * Builder for {@link BlogPostForMailItem}
+     */
+    public static class Builder {
 
-    private String determineAuthorLabel(Blog blogger) {
-        if (blogger.getTwitter() != null) {
-            return "<a href=\"https://twitter.com/" + blogger.getTwitter().substring(1) + "\">" + blogger.getAuthor() + "</a>";
-        } else {
-            return blogger.getAuthor();
+        private final BlogPostForMailItem instance = new BlogPostForMailItem();
+
+        public Builder from(BlogPost blogPost) {
+            instance.title = blogPost.getTitle();
+            instance.url = blogPost.getUrl();
+            instance.authorLabel = determineAuthorLabel(blogPost.getBlog());
+            return this;
+        }
+        
+        public Builder withTitle(String title) {
+            instance.title = title;
+            return this;
         }
-    }
 
+        public Builder withUrl(String url) {
+            instance.url = url;
+            return this;
+        }
+        
+        public Builder withAuthorLabel(Blog blog) {
+            instance.authorLabel = determineAuthorLabel(blog);
+            return this;
+        }
+        
+        public Builder withUrlParameter(String name, String value) {
+            Preconditions.checkState(
+                instance.url != null,
+                "Url could not be null. Please set url first"
+            );
+            
+            instance.url = UriComponentsBuilder
+                .fromHttpUrl(instance.url)
+                .queryParam(name, value)
+                .build().toString();
+            
+            return this;
+        }
+        
+        public Builder withDefaultUTMParameters() {
+            withUrlParameter(UTMParameters.UTM_SOURCE_KEY, UTM_SOURCE);
+            withUrlParameter(UTMParameters.UTM_MEDIUM_KEY, UTM_MEDIUM);
+            withUrlParameter(UTMParameters.UTM_CAMPAING_KEY,
+                String.format(
+                    "%s#%s", UTM_SOURCE,
+                    new SimpleDateFormat("yyyy-mm-dd").format(new Date())
+                )
+            );
+            
+            return this;
+        }
+        
+        public BlogPostForMailItem build() {
+            Preconditions.checkState(instance.title != null, "Tittle cannot be null");
+            Preconditions.checkState(instance.url != null, "Url cannot be null");
+            Preconditions.checkState(instance.authorLabel != null, "Author cannot be null");
+            
+            return instance;
+        }
+
+        private String determineAuthorLabel(Blog blogger) {
+            if (blogger.getTwitter() != null) {
+                return String.format(
```

Comment:
Because `withUrlParameter(UTMParameters.UTM_CAMPAING_KEY,String.format("%s#%s", UTM_SOURCE, new SimpleDateFormat("yyyy-mm-dd").format(new Date())));` is too long in my opinion. IDE's built-in code formatters tend to breaks the line in very strange places like:
```java
Blog newBlog = new Blog(bloggerEntry.getJsonId(), bloggerEntry.getName(),
                StringUtils.lowerCase(bloggerEntry.getRss()), bloggerEntry.getTwitter(), nowProvider.now(), bloggerEntry.getBlogType());
```
I used to format code that way to improve readability. I would rather format this like that:
```java
Blog newBlog = new Blog(
    bloggerEntry.getJsonId(), bloggerEntry.getName(),
    StringUtils.lowerCase(bloggerEntry.getRss()),
    bloggerEntry.getTwitter(), nowProvider.now(),
    bloggerEntry.getBlogType()
);
```
 For more details please see http://www.yegor256.com/2014/10/23/paired-brackets-notation.html (Im not following these rules in any case). I know that nowadays we have bigger displays but personally I'm trying to keep my code lines no longer than 80-100 characters. If line has more characters Im trying to break It in a reasonable place. Moreover, if code is hard to format it sometimes shows problems with design e.g breaking single responsability. Nevertheless, I can stop formatting code that way and relay on IDE formatter if you like.